### PR TITLE
refactoring for torque based driving

### PR DESCRIPTION
**Pull request by shayana18**
### Changelist 
- changed power based driving to torque based

### Testing Done

- [x] Compiles
- [x] Runs
- [x] Tested on track

### CODE REVIEW

**Review by 1mackenziekyle — COMMENTED** [review]
Instead of calculating the power request inside app_torqueVectoring.c, we should instead just ensure that the requested torque is within the power limit. 

image

In lines 16 and 17 above, replace `torque_lim_Nm` with your requested torque, which should be calculated in app_TorqueVectoring_handleAcceleration(), and added to ActiveDifferential_Inputs. 

then do the same thing thats being done here: 

image

but with `torque_lim_Nm`

**Comment by shayana18** [comment]
@1mackenziekyle. I get what you're saying. Take a look now

File: firmware/quadruna/VC/src/app/vehicle_dynamics/app_activeDifferential.c [hunk 1]
```diff
@@ -13,14 +13,14 @@ void app_activeDifferential_computeTorque(ActiveDifferential_Inputs *inputs, Act
...
+    float torque_left_Nm  = inputs->requested_torque * (1 + Delta);
+    float torque_right_Nm = inputs->requested_torque * (1 - Delta);
     float torque_max_Nm   = fmaxf(torque_left_Nm, torque_right_Nm);
 
     float scale = 1.0f;
-    if (torque_max_Nm > MAX_TORQUE_REQUEST_NM)
+    if (torque_max_Nm > torque_lim_Nm)
     {
-        scale = MAX_TORQUE_REQUEST_NM / torque_max_Nm;
+        scale = torque_lim_Nm / torque_max_Nm;
```
Comment (philam2001) [inline firmware/quadruna/VC/src/app/vehicle_dynamics/app_activeDifferential.c:23]: Ah I guess that would affect this too.

File: firmware/quadruna/VC/src/app/vehicle_dynamics/app_torqueVectoring.c [hunk 2]
```diff
@@ -39,6 +40,9 @@ static float current_consumption;
 static float left_motor_temp_C;
 static float right_motor_temp_C;
 static float steering_angle_deg;
+static float left_pedal_power_request;
```
Comment (philam2001) [inline firmware/quadruna/VC/src/app/vehicle_dynamics/app_torqueVectoring.c:43]: are you using these anywhere?

File: firmware/quadruna/VC/src/app/vehicle_dynamics/app_torqueVectoring.c [hunk 3]
```diff
@@ -1,3 +1,4 @@
+#include <math.h>
```
Comment (philam2001) [inline firmware/quadruna/VC/src/app/vehicle_dynamics/app_torqueVectoring.c:1]: This is already included pretty sure through another one of the .h files

**Review by philam2001 — COMMENTED** [review]
looks good, just make sure to satisfy all the ci rules i think you need to format it.

**Comment by philam2001** [comment]
Fix your tests and then approved

File: firmware/quadruna/VC/src/app/vehicle_dynamics/app_torqueVectoring.c [hunk 4]
```diff
@@ -39,6 +39,7 @@ static float current_consumption;
 static float left_motor_temp_C;
 static float right_motor_temp_C;
 static float steering_angle_deg;
+static float min_pedal_power_request;
```
Comment (philam2001) [inline firmware/quadruna/VC/src/app/vehicle_dynamics/app_torqueVectoring.c:42]: what is this used for?

**Review by philam2001 — APPROVED** [review]

**Pull request merged**